x-pack/heartbeat/monitors/browser/synthexec/synthexec.go

```diff
@@ -26,8 +30,82 @@ import (
 
 const debugSelector = "synthexec"
 
+// ensure compatability of synthetics by enforcing the installed
+// version never goes beyond this range
+const ExpectedSynthVersion = "<2.0.0"
+
+type PackageJSON struct {
+	Dependencies struct {
+		SynthVersion string `json:"@elastic/synthetics"`
+	} `json:"dependencies"`
+	DevDependencies struct {
+		SynthVersion string `json:"@elastic/synthetics"`
+	} `json:"devDependencies"`
+}
+
+var numberRegex = regexp.MustCompile("[^0-9]+")
+
+// parsed a given dep version by ignoring all range tags (^, = , >, <)
+func parseVersion(version string) string {
+	dotParts := strings.SplitN(version, ".", 4)
+	parsed := []string{}
+
+	for _, v := range dotParts[:3] {
+		value := numberRegex.ReplaceAllString(v, "")
+		parsed = append(parsed, value)
+	}
+	return strings.Join(parsed, ".")
+}
+
+func validateVersion(expected string, current string) error {
+	expectedRange, err := semver.NewConstraint(expected)
+	if err != nil {
+		return err
+	}
+
+	parsed := parseVersion(current)
+	currentVersion, err := semver.NewVersion(parsed)
+	if err != nil {
+		fmt.Println("error parsing synthetics version: %w", err)
+	}
+
+	isValid := expectedRange.Check(currentVersion)
+	if !isValid {
+		return fmt.Errorf("provided synthetics version %s is not compatible", current)
+	}
+	return nil
+}
+
+func validatePackageJSON(path string) error {
+	pkgData, err := ioutil.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("could not read file:%s %w", path, err)
```

Comment:
I'd clean this up a bit so it's clear what the file path is and what the error is. I tend to format these things like:

`fmt.Errorf("could not read file '%s': %w", path, err)`

The quotes really helps when the file is an empty value